gqcp/include/Basis/Transformations/SimpleOrbitalRotationGenerators.hpp

```diff
@@ -0,0 +1,146 @@
+ * 
+ *  @tparam _Scalar                                           The scalar type used for a orbital rotation generator: real or complex.
+ *  @tparam _DerivedOrbitalRotationGenerator                  The type of the orbital rotation generator that derives from this class, enabling CRTP and compile-time polymorphism.
+ */
+template <typename _Scalar, typename _DerivedOrbitalRotationGenerators>
+class SimpleOrbitalRotationGenerators {
+
+public:
+    // The scalar type used for a orbital rotation generator: real or complex.
+    using Scalar = _Scalar;
+
+    // The type of the orbital rotation generator that derives from this class, enabling CRTP and compile-time polymorphism.
+    using DerivedOrbitalRotationGenerators = _DerivedOrbitalRotationGenerators;
+
+private:
+    // The number of orbitals (spinors for the general(ized) case, spin-orbitals for the restricted and unrestricted case) that can be rotated using these orbital rotation generators.
+    size_t number_of_orbitals;
+
+    // The strict lower triangle of the kappa matrix.
+    VectorX<Scalar> v;
+
+
+public:
+    /**
+     *  MARK: Constructors
+     */
+
+    /**
+     *  Create a `SimpleOrbitalRotationGenerator' from a given vector containing orbital rotation generators kappa_PQ with P>Q.
+     * 
+     *  @param  v        The orbital rotation generators represented as a vector that corresponds to the strict lower triangle of the kappa matrix (kappa_PQ with P>Q).
+     */
+    SimpleOrbitalRotationGenerators(const VectorX<Scalar>& v) :
+        v {v},
+        number_of_orbitals {strictTriangularRootOf(v.size())} {}
+
+
+    /**
+     *  Create a `SimpleOrbitalRotationGenerator' from a given kappa matrix.
+     * 
+     *  @param  kappa_matrix        The orbital rotation generators represented as a vector that corresponds to the full anti-Hermitian the kappa matrix.
+     */
+    SimpleOrbitalRotationGenerators(const SquareMatrix<Scalar>& kappa_matrix) :
+        SimpleOrbitalRotationGenerators(kappa_matrix.pairWiseStrictReduced()) {}
+
+
+    /**
+     *  MARK: Named constructors
+     */
+
+    /**
+     *  Construct orbital rotation generators by adding redundant (i.e. 0) generators to the given occupation_type - occupation_type generators.
+     * 
+     *  @param generators                           The orbital rotation generators of the specified ocupation types.
+     *  @param row_occupation_type                  The occupation type of the rows of the orbital rotation generator kappa matrix.
+     *  @param column_occupation_type               The occupation type of the column of the orbital rotation generator kappa matrix.
+     *  @param K                                    The total number of orbitals. In the general(ized) case these are spinors, for restricted these will be spin-orbitals.
```

Comment:
I didn't mention the unrestricted case here, because the `UOrbitalRotationGenerators` have their own version of the API, but I'll add it as suggested!